vm/src/obj/objlist.rs

```diff
@@ -409,6 +409,26 @@ fn list_pop(vm: &mut VirtualMachine, args: PyFuncArgs) -> PyResult {
     }
 }
 
+fn list_remove(vm: &mut VirtualMachine, args: PyFuncArgs) -> PyResult {
+    arg_check!(
+        vm,
+        args,
+        required = [(list, Some(vm.ctx.list_type())), (needle, None)]
+    );
+
+    let mut elements = get_mut_elements(list);
+    let i = elements
+        .iter()
+        .position(|r| objbool::get_value(&vm._eq(needle.clone(), r.clone()).unwrap()));
```

Comment:
I think here we should use `?` rather than unwrapping.

Reply:
If I change the `unwrap` to `?` then I got the error:

```
cannot use the `?` operator in a function that returns `bool`
```

Reply:
Yes, that's because `?` can only be used in a function that returns a `Result`. `list_remove` does, but you are calling it inside the closure argument to `position` which returns a `bool`. The solution is probably to convert the iterator chain to a for loop, similar to [`list_index`](https://github.com/RustPython/RustPython/blob/2ac195e539cfe30332cfee356473f7c965e3cfe1/vm/src/obj/objlist.rs#L257-L272).

Reply:
Thank you for explanation. But if I'm trying remove the element inside the for loop in like in `list_index` that I have some borrowing error.

Reply:
Yes, mutating inside the loop would be iterator invalidation, so naturally rust disallows it. You'll need to introduce a local binding, so that you can remove the item after breaking the loop.

Reply:
Ok. In the last commit I change it and now first I searching for index and after that remove the element.